WalletWasabi/WabiSabi/Client/BobClient.cs

```diff
@@ -1,30 +1,41 @@
 using NBitcoin;
 using System;
+using System.Collections.Generic;
+using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using WalletWasabi.Crypto.ZeroKnowledge;
 
 namespace WalletWasabi.WabiSabi.Client
 {
 	public class BobClient
 	{
-		public BobClient(uint256 roundId, ArenaClient arenaClient)
+		public BobClient(uint256 roundId, ArenaClient arenaClient, IEnumerable<Credential> realAmountCredentials, IEnumerable<Credential> realVsizeCredentials)
 		{
 			RoundId = roundId;
 			ArenaClient = arenaClient;
+			RealAmountCredentials = realAmountCredentials.ToArray();
+			RealVsizeCredentials = realVsizeCredentials.ToArray();
 		}
 
 		private uint256 RoundId { get; }
 		private ArenaClient ArenaClient { get; }
+		private Credential[] RealAmountCredentials { get; set; }
+		private Credential[] RealVsizeCredentials { get; set; }
 
 		public async Task RegisterOutputAsync(Money amount, Script scriptPubKey, CancellationToken cancellationToken)
 		{
-			await ArenaClient.RegisterOutputAsync(
+			// TODO: what to do with the credentials returned?
```

Comment:
Discard and leave a TODO comment to remove from the code after backend is updated.

Based on the call I think 2 weeks ago I implemented the credential dependency planner to always give output registration nodes an exact input balance. This means we can remove them from the backend.

Note that although this saves CPU resources by removing 2 range proofs per output, we will still pay at least some of the computational cost with reissuance cover traffic, which is necessary for privacy with this approach.

Reply:
Quickly though it through, we don't need it? Only exact output registration should be made, so there is nothing valuable to return. Another PR.